layers/+lang/haskell/packages.el

```diff
@@ -12,27 +12,61 @@
 (setq haskell-packages
   '(
     cmm-mode
+    company
     (company-cabal :requires company)
-    company-ghci
-    company-ghc
+
+    ;; ghci completion backend
+    (company-ghci
+     :requires company
+     :toggle (eq haskell-completion-backend 'ghci))
+
+    ;; ghc-mod completion backend
+    (company-ghc
+     :requires company
+     :toggle (eq haskell-completion-backend 'ghc-mod))
+    (ghc :toggle (eq haskell-completion-backend 'ghc-mod))
+
+    ;; intero completion backend
+    (intero
+     :requires company
+     :toggle (eq haskell-completion-backend 'intero))
+
+    ;; dante completion backend
+    (dante
+     :requires company
+     :toggle (and (version<= "25" emacs-version)
+                  (eq haskell-completion-backend 'dante)))
+
     flycheck
     (flycheck-haskell :requires flycheck)
     ggtags
-    ghc
     haskell-mode
     haskell-snippets
     helm-gtags
     (helm-hoogle :requires helm)
     hindent
     hlint-refactor
-    intero
-    (dante :toggle (version<= "25" emacs-version))
     ))
 
 (defun haskell/init-cmm-mode ()
   (use-package cmm-mode
     :defer t))
 
+(defun haskell/post-init-company ()
```

Comment:
Why do you call `spacemacs|add-company-backends` here instead of the `haskell-init-company-*backend*` functions?